bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdf_pyz.py

```diff
@@ -0,0 +1,277 @@
+import re
+from .._numbadeclare import _NumbaDeclareDecorator
+
+# This class is still to undergo some restructuring thus the docstring may not be complete.
+# The class will be fully realized after the Define has been Pythonized.
+class FunctionJitter:
+    """
+    Class to jit function using NumbaDeclare.
+    It takes a python callable (or expression), an RDF and optionally a column list and dictionary containing any other arguments for the callable.
+    It determines the function signature and Declares the function and returns the corresponding function call.
+
+    Attributes
+    ----------
+    rdf: RDataFrame object on which the function's column arguments are in
+    col_names: The list of columns in rdf 
+    func: The python callable that is to be jitted.
+    params: List of parameters of the above callable
+    func_args: Dict to store the value of each param
+    args_info: Information about the type and value of each param
+    func_sign: Contains the function signature
+    func_call: Contains the function call as a string.
```

Comment:
Missing attribute definition in the `__init__` method and type hints

Reply:
An example would be
```python
class Foo:

    def __init__(self):
        self.a: str

    def seta_str(self):
        self.a = "hello"

    def seta_int(self):
        self.a = 10
```

The instance variable `a` is annotated in the `__init__` method. Notice that you don't need to assign it with a default value. But functions that later initialize that attribute are properly type-checked by e.g. `mypy`:

```
$: mypy --check-untyped-defs --pretty test.py 
test.py:10: error: Incompatible types in assignment (expression has type "int", variable has type "str")
            self.a = 10
                     ^
Found 1 error in 1 file (checked 1 source file)
```